kip-0031.md

```diff
@@ -0,0 +1,107 @@
+---
+KIP: "0031"
+Title: fungible-meta-v1
+Author: CryptoPascal  
+Status: Early Draft
+Type: Standard
+Category: Chainweb / Fungible
+Created: 2024-09-28
+---
+
+## Abstract
+
+Define and implement a new way for tokens creators to export their Meta-Data to Wallets, Frontends, and Other services.
+While this is currently handled off-chain, this proposal is intended to take a an "on-chain" approach.
+
+
+## Rationale
+
+While `fungible-v2` provides technical functions to allow "technical on-chain" interoperability between tokens, meta-data and information about fungible tokens are missing.
+
+Currently, listing tokens for Wallet, Frontends, Exchanges, and other third parties is a cumbersome process. Maintainers have to manually contact tokens issuers to retrieve logos, names, and other useful information to display to users. This ends up with disparities between different wallets or frontends. Users complain that they don't have reliable information about their investment.
+
+Moreover, token issuers have no clean and defined way to inform wallets when some changes happen with their token (Rebrand, new issuance, ...)
+
+And finally, since Kadena is multichains, wallets need to be informed on how to use the token, regarding chains deployment:
+  - Some tokens may not be available on all chains.
+  - Some tokens may have a single issuance chain where supply is accounted
+
+The proposal is **to create an optional interface that can be implemented by token modules** to automatically inform third parties about their meta-data, and displayable information to users. Wallets, frontends, ... are expected to make local calls to the `kip.fungible-meta` implemented functions.
+
+#### Trustability
+
+Information (especially supply) relies **only** on token issuers and module writers good faith. Clients should not blindly trust them. Sorting legitimate tokens from illegitimate ones is not in the scope of this KIP.
+
+#### Non-Obligability
+For being compliant with this KIP, token issuers have no obligation to provide all data.
+And clients (Wallets, Frontends), have no obligation to show everything.
+
+#### Supply considerations
+
+One important piece of information required by investors is the total and circulating supply.
+Kadena being multichain, when dynamic minting/burning tokens are considered, a new issue arises:
+  - Usually, minting / burning / emission happens on a single chain.
+  - **But other chains are not aware**: they just receive X-chain transfers.
+
+As a consequence, it's highly probable that supply information won't be accurate on most chains. That's why this KIP proposes the concept of **main-chain** => The module itself informs the client where an accurate value of supply amounts can be found.
```

Comment:
Supply will be accurate on individual chains if there as a chain-level supply noted in the interface for a token's metadata, just not a "total supply" across all the network. I think here, we need to separate the idea of what is _autonomous_ metadata, as in facts that a ledger can keep about itself, vs _dependent_ metadata - facts that require some kind of oracle to contribute. A main chain being an example of this oracle, but not the only valid kind out there.